acceptance/tests/resource/file/source_attribute.rb

```diff
@@ -1,282 +1,283 @@
+      dir_to_rm = host['platform'] =~ /windows/ ? @target_dir_on_windows : @target_dir_on_nix
 
-    checksums.each do |checksum_type|
-      on(host, "rm #{file_to_rm}#{checksum_type}", :acceptable_exit_codes => [0,1])
-      on(host, "rm -r #{dir_to_rm}#{checksum_type}", :acceptable_exit_codes => [0,1])
+      checksums.each do |checksum_type|
+        on(host, "rm #{file_to_rm}#{checksum_type}", :acceptable_exit_codes => [0,1])
+        on(host, "rm -r #{dir_to_rm}#{checksum_type}", :acceptable_exit_codes => [0,1])
+      end
     end
   end
-end
-
-step "Setup - create environment and test module"
-# set directories
-testdir = master.tmpdir('file_source_attr')
-env_dir = "#{testdir}/environments"
-prod_dir = "#{env_dir}/production"
-manifest_dir = "#{prod_dir}/manifests"
-manifest_file = "#{prod_dir}/manifests/site.pp"
-module_dir = "#{prod_dir}/modules"
-test_module_dir = "#{module_dir}/source_test_module"
-test_module_manifests_dir = "#{test_module_dir}/manifests"
-test_module_files_dir = "#{test_module_dir}/files"
-mod_manifest_file = "#{test_module_manifests_dir}/init.pp"
-mod_source_file = "#{test_module_files_dir}/source_file"
-mod_source_dir = "#{test_module_files_dir}/source_dir"
-mod_source_dir_file = "#{mod_source_dir}/source_dir_file"
-
-mod_source = ' the content is present'
-
-def mod_manifest_entry(checksum_type = nil)
-  checksum = if checksum_type then "checksum => #{checksum_type}," else "" end
-  manifest = <<EOF
-  $target_file#{checksum_type} = $::kernel ? {
-    \\'windows\\' => \\'#{@target_file_on_windows}#{checksum_type}\\',
-    default   => \\'#{@target_file_on_nix}#{checksum_type}\\'
-  }
-
-  file { $target_file#{checksum_type}:
-    source => \\'puppet:///modules/source_test_module/source_file\\',
-    #{checksum}
-    ensure => present
-  }
-
-  $target_dir#{checksum_type} = $::kernel ? {
-    \\'windows\\' => \\'#{@target_dir_on_windows}#{checksum_type}\\',
-    default   => \\'#{@target_dir_on_nix}#{checksum_type}\\'
-  }
-
-  file { $target_dir#{checksum_type}:
-    source => \\'puppet:///modules/source_test_module/source_dir\\',
-    #{checksum}
-    ensure => directory,
-    recurse => true
-  }
-EOF
-  manifest
-end
 
-mod_manifest = <<EOF
-class source_test_module {
-#{checksums.collect { |checksum_type| mod_manifest_entry(checksum_type) }.join("\n")}
-}
-EOF
-
-env_manifest = <<EOF
-filebucket { \\'main\\':
-  server => \\'#{master}\\',
-  path   => false,
-}
-
-File { backup => \\'main\\' }
-
-node default {
-  include source_test_module
-}
-EOF
-
-# apply manifests to setup environment and modules
-apply_manifest_on(master, <<-MANIFEST, :catch_failures => true)
-  File {
-    ensure => directory,
-    mode => '0755',
-  }
-
-  file {
-    '#{testdir}':;
-    '#{env_dir}':;
-    '#{prod_dir}':;
-    '#{manifest_dir}':;
-    '#{module_dir}':;
-    '#{test_module_dir}':;
-    '#{test_module_manifests_dir}':;
-    '#{test_module_files_dir}':;
-  }
+  step "Setup - create environment and test module"
```

Comment:
i'd like to see closures on `step` as well as `test_name` but it shouldn't hold up this PR